lib/ipam/interface.go

```diff
@@ -36,7 +36,7 @@ type Interface interface {
 	// which is useful for dataplanes that need to know the subnet (such as Windows).
 	//
 	// In case of error, returns the IPs allocated so far along with the error.
-	AutoAssign(ctx context.Context, args AutoAssignArgs) ([]cnet.IPNet, []cnet.IPNet, error)
+	AutoAssign(ctx context.Context, args AutoAssignArgs) ([]cnet.IPNet, []cnet.IPNet, *IPAMAssignmentInfo, *IPAMAssignmentInfo, error)
```

Comment:
We should update the interface comment to explain the new return values

Reply:
Another nice simplification might be to remove the `[]cnet.IPNet` return values altogether and embed them in the returned struct instead. This reduces the number of return values, as well as the number of places we're tracking the same information (e.g., num assigned IPs).

Maybe could call the struct `IPAMResult`

Reply:
Will do!